samtranslator/model/sam_resources.py

```diff
@@ -867,8 +868,123 @@ def to_cloudformation(self, **kwargs):
+
+            # get the alarm name and the properties the user defined for the alarm
+            alarm_name = list(dict_item.keys())[0]
+            alarm_item = dict_item[alarm_name]
+
+            # MetricName property is required
+            if alarm_item is None or "MetricName" not in alarm_item:
+                raise InvalidResourceException(
+                    self.logical_id,
+                    "CloudWatch alarm '{key}' is missing required property 'MetricName'.".format(key=alarm_name),
+                )
+
+            metric_name = alarm_item["MetricName"]
+            valid_metrics = ["SuccessPercent", "Failed", "Duration"]
+
+            # MetricName must be one of "SuccessPercent", "Failed" and "Duration"
+            if metric_name not in valid_metrics:
+                raise InvalidResourceException(
+                    self.logical_id,
+                    "MetricName needs to be one of {}".format(valid_metrics),
+                )
+
+            # make sure all the alarm names are unique
+            if alarm_name in list_of_alarm_names:
+                raise InvalidResourceException(self.logical_id, "Duplicate CloudWatch alarm names")
+            else:
+                list_of_alarm_names.append(alarm_name)
+
+    def _construct_cloudwatch_alarms(self, dict_item):
+        """Constructs an CloudWatch::Alarm resource if the user specifies the CloudWatchAlarm property in Serverless Canary
+
+        :returns: the generated CloudWatch Alarm
+        :rtype: model.cloudwatch.CloudWatchAlarm
+        """
+
+        # The default values for ComparisonOperator, Threshold and Period based on the MetricName provided by the user
+        # These default values were acquired from the Create Canary page in the Synthetics Canary dashboard
+        default_values = {
+            "SuccessPercent": {"ComparisonOperator": "LessThanThreshold", "Threshold": 90, "Period": 300},
+            "Failed": {"ComparisonOperator": "GreaterThanOrEqualToThreshold", "Threshold": 1, "Period": 300},
+            "Duration": {"ComparisonOperator": "GreaterThanThreshold", "Threshold": 30000, "Period": 900},
+        }
+
+        alarm_name = list(dict_item.keys())[0]
+        alarm_item = dict_item[alarm_name]
+
+        cloudwatch_alarm = CloudWatchAlarm(
+            logical_id=alarm_name,
+            depends_on=self.depends_on,
+            attributes=self.get_passthrough_resource_attributes(),
+        )
+
+        # default settings for the CloudWatch alarms
+        # the settings are identical to the Alarms that are made by Synthetics Canary using their dashboard
+        cloudwatch_alarm.MetricName = alarm_item["MetricName"]
+        cloudwatch_alarm.Namespace = "CloudWatchSynthetics"
+        cloudwatch_alarm.EvaluationPeriods = 1
+        cloudwatch_alarm.Statistic = "Sum"
+        cloudwatch_alarm.TreatMissingData = "notBreaching"
+        cloudwatch_alarm.Dimensions = [{"Name": "CanaryName", "Value": {"Ref": self.logical_id}}]
```

Comment:
I understand that we decided to make these values fixed as done by Synthetics console, but I am wondering if there is value in simply making them passthrough with default? Thoughts?

Reply:
Namespace and Statistic are required for an alarm with MetricName and Dimensions is needed to connect the alarm to the metric of a specific canary (rather than the metric across all canaries). Also EvaluationPeriods is a required property for all alarms. The only one that can be passed through with the default is TreatMissingData